### PR TITLE
added updated version of showEveryNthPerson

### PR DESCRIPTION
**Pull request by sanjivsahayamrea**
So I've added some extra validation rules and beefed up the tests to cover almost all cases.

### CODE REVIEW

**Comment by wjlow** [comment]
Can you rename the test? To something like "should show every Nth person"

I wonder if it's a little late to add this for tomorrow. A bit hesitant to merge this because I'm not sure the presenters are ready with the solution that'll meet all the test cases.

**Comment by sanjivsahayamrea** [comment]
yep, happy to leave this for next time :)

**Comment by ssanj** [comment]
I've updated this exercise as I think we can trial it out on the study group. I think it was handy that we taught then Tuples, as this exercise now makes more sense. I have also extracted out the `zipWithIndex` bit into its own exercise.

**Comment by ssanj** [comment]
#158

File: src/main/scala/fundamentals/level02/ListExercises.scala [hunk 1]
```diff
@@ -166,4 +166,50 @@ object ListExercises {
...
+    * Hint: Use `zipWithIndex`
+    */
+  def personWithIndex(people: List[Person]): List[(Person, Int)] = ???
+
+  /**
+    * Log every nth person from the `peopleList` given an index `n`.
+    *
+    * scala> showEveryNthPerson(2, peopleList)
+    * = List("Karen Page is 27 years old", "Claire Temple is 32 years old")
+    *
```
Comment (ashokkumar) [inline src/main/scala/fundamentals/level02/ListExercises.scala]: I think the output would be `List("Karen Page is 27 years old", "Claire Temple is 32 years old","Elektra Natchios is 29 years old")`

Comment (sanjivsahayamrea) [reply src/main/scala/fundamentals/level02/ListExercises.scala]: oh yeah, I added a 6th person and forgot to update the list.

**Comment by sanjivsahayamrea** [comment]
@ashokkumar I made some updates and rewrote the `youngestPerson` exercise to reuse `peopleList`

**Comment by ashokkumar** [comment]
All looks good @sanjivsahayamrea

**Pull request merged**